File: iosApp/iosAppTests/CompanionObject.swift
//
//  sdf.swift
//  iosAppTests
//
//  Created by Nicola De Fiorenze on 03/07/18.
//  Copyright © 2018 Nicola De Fiorenze. All rights reserved.
//

import XCTest
import KotlinLibrary

class CompanionObject: XCTestCase {
    
    func testCompanionObjectUsage() {
        XCTAssertEqual(12, KotlinLibraryContainerCompanion().my_SUPER_NICE_CONST)
        let firstContainer = KotlinLibraryContainer(state: "first")
        let secondContainer = KotlinLibraryContainer(state: "second")

        XCTAssertFalse(firstContainer === secondContainer)

        KotlinLibraryContainerCompanion().sharedVar = "Hello"

        XCTAssert(firstContainer.getDecoratedState() == "Hello first")
        XCTAssert(secondContainer.getDecoratedState() == "Hello second")
    }

}

File: androidApp/app/src/test/java/com/ndefiorenze/samplemultiplatformkotlinapp/CompanionObject.kt
package com.ndefiorenze.samplemultiplatformkotlinapp

import com.ndefiorenze.Container
import org.junit.Assert
import org.junit.Test

class CompanionObject {

    @Test
    fun usageOfCompanionObject() {
        Assert.assertEquals(12, Container.MY_SUPER_NICE_CONST)
        val firstContainer = Container("first")
        val secondContainer = Container("second")

        Assert.assertFalse(firstContainer === secondContainer)

        Container.sharedVar = "Hello"

        Assert.assertEquals("Hello first", firstContainer.getDecoratedState())
        Assert.assertEquals("Hello second", secondContainer.getDecoratedState())
    }

}